### ISSUE TITLE
Domain Type implementations needed by IBC

### ISSUE THREAD

**Issue by ancazamfir** [informalsystems/tendermint-rs#634]
Following domain types and `TryFrom/ From` implementations are needed for (de)serialization of client IBC messages:

`tendermint_proto::types::SignedHeader` <===> `tendermint::block::signed_header::SignedHeader`
`tendermint_proto::types::ValidatorSet` <===> `tendermint::validator::Set`
`tendermint_proto::abci::ConsensusParams` <===> `tendermint::consensus::params::Params`

Required for https://github.com/informalsystems/ibc-rs/issues/277

**Comment by greg-szabo** [comment]
I'm working on this but it's going to take a bit more time than I expected. All these types have fields with other types that also need to be created. I'll try to measure how much this work is, so I can give more appropriate status updates.

**Comment by greg-szabo** [comment]
Types to implement: (indentation shows dependency on the type above)
```
SignedHeader
  block::Header
    ConsensusVersion
  block::Commit (fix: Round is still u32 here for some reason)
    CommitSigs (what is the purpose of this? It's not doing any validation or filtering)
      CommitSig (fix: figure out what to do with the RawCommitSig implemented for serde)

validator::Set
  Info
  vote::Power
  ProposerPriority

consensus::params::Params
  block::Size
  evidence::Params
    evidence::Duration
  evidence::ValidatorParams
    public_key::Algorithm
```

I haven't found any proof that these were amino-encoded before. If these types are broken in the JSON encoding, then this is something we haven't encountered before and we were only vaguely referring to "fixing JSON-encoding" in the v0.34 upgrade.

Upgrading JSON-encoding was supposed to be a later little project where we harmonize the validation given by domain types with the JSON validation. It seems we'll have to prioritize this.

I'm going to get back to this PR as soon as I have a better picture what a "JSON-harmonization" would entail.

**Comment by greg-szabo** [comment]
`SignedHeader` and `validator::Set` are implemented on the greg/research-json branch.

**Comment by greg-szabo** [comment]
All three are now implemented in #639 . Some of it was unused before, so if you find any issues we can fix / change the structs in followup issues.
